### DISCUSSION TITLE
Example of infinite scroll with useSWRInfinite

### DISCUSSION THREAD

**Discussion by 4ndv** [vercel/swr#704] [Q&A] [answered]
I'm trying to build scroll-based lazyloading, but always get stuck with the same problem: next page is fetched before previous rendered and in result - next page never appears. I'm already wasted almost 6 hours and cannot make it work. Does someone have any examples of doing so?

**Comment by 4ndv** [comment, marked as answer]
In case anyone stuck on the same stuff as I am, I was able to make it work with this component: https://www.npmjs.com/package/react-infinite-scroll-component

**Reply by ezeikel** [reply]
Do you have a code sample of how you got the two to play nice? Trying to do the same thing

**Reply by abdulazizalmalki-gh** [reply]
But this does not use SWR

**Comment by ramyareye** [comment]
related example added [here](https://github.com/vercel/swr/tree/master/examples/infinite-scroll)

**Reply by JanderSilv** [reply]
Thank you so much.

**Reply by kocaagasefa** [reply]
> related example added [here](https://github.com/vercel/swr/tree/master/examples/infinite-scroll)

Hi. This example does not work properly in one case.
When fetched page is small and loading element keeps  visible on screen after fetch, it does not trigger fetching next page.
You can check by zooming out browser

```js
is visible - true
is reacthing end - false
is refreshing - false
```

![image](https://user-images.githubusercontent.com/32370381/215189947-2a23eacf-2381-4c25-9d6e-9aa5c2837494.png)

**Reply by huksley** [reply]
@ramyareye Do you have example where it is actually loads when you scroll long content? This would very useful!

**Reply by kocaagasefa** [reply]
https://stackblitz.com/edit/github-wyn3fz?file=pages%2Findex.js

Issue occurs when setting `perPage` to a small value (like 5)

![image](https://user-images.githubusercontent.com/32370381/218262048-b73a8770-5176-431a-b091-526e6cfe096b.png)

**Reply by ramyareye** [reply]
> @ramyareye Do you have example where it is actually loads when you scroll long content? This would very useful!

I don't get it, I can check if u provide a codesandbox

**Reply by ramyareye** [reply]
> https://stackblitz.com/edit/github-wyn3fz?file=pages%2Findex.js
> 
> Issue occurs when setting `perPage` to a small value (like 5)
> 
> ![image](https://user-images.githubusercontent.com/32370381/218262048-b73a8770-5176-431a-b091-526e6cfe096b.png)

It works probably for me with 5

If doesn't for u, probably should play with isVisible and isLoading inside useEffect
